apps/studio/components/interfaces/QueryPerformanceV2/QueryPerformance.tsx

```diff
@@ -0,0 +1,201 @@
+import ShimmeringLoader from 'ui-patterns/ShimmeringLoader'
+
+interface QueryPerformanceProps {
+  queryHitRate: PresetHookResult
+  queryPerformanceQuery: DbQueryHook<any>
+}
+
+export const QueryPerformance = ({
+  queryHitRate,
+  queryPerformanceQuery,
+}: QueryPerformanceProps) => {
+  const router = useRouter()
+  const { preset } = useParams()
+  const { project } = useProjectContext()
+  const [page, setPage] = useState<
+    | QUERY_PERFORMANCE_REPORT_TYPES.MOST_TIME_CONSUMING
+    | QUERY_PERFORMANCE_REPORT_TYPES.MOST_FREQUENT
+    | QUERY_PERFORMANCE_REPORT_TYPES.SLOWEST_EXECUTION
+  >(
+    (preset as QUERY_PERFORMANCE_REPORT_TYPES) ?? QUERY_PERFORMANCE_REPORT_TYPES.MOST_TIME_CONSUMING
+  )
+  const [showResetgPgStatStatements, setShowResetgPgStatStatements] = useState(false)
+
+  const handleRefresh = async () => {
+    queryPerformanceQuery.runQuery()
+    queryHitRate.runQuery()
+  }
+
+  const { data: mostTimeConsumingQueries, isLoading: isLoadingMTC } = useQueryPerformanceQuery({
+    preset: 'mostTimeConsuming',
+  })
+  const { data: mostFrequentlyInvoked, isLoading: isLoadingMFI } = useQueryPerformanceQuery({
+    preset: 'mostFrequentlyInvoked',
+  })
+  const { data: slowestExecutionTime, isLoading: isLoadingMMF } = useQueryPerformanceQuery({
+    preset: 'slowestExecutionTime',
+  })
+
+  const QUERY_PERFORMANCE_TABS = [
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.MOST_TIME_CONSUMING,
+      label: 'Most time consuming',
+      description: 'Lists queries ordered by their cumulative total execution time.',
+      isLoading: isLoadingMTC,
+      max:
+        (mostTimeConsumingQueries ?? []).length > 0
+          ? Math.max(...(mostTimeConsumingQueries ?? []).map((x: any) => x.total_time)).toFixed(2)
+          : undefined,
+    },
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.MOST_FREQUENT,
+      label: 'Most frequent',
+      description: 'Lists queries in order of their execution count',
+      isLoading: isLoadingMFI,
+      max:
+        (mostFrequentlyInvoked ?? []).length > 0
+          ? Math.max(...(mostFrequentlyInvoked ?? []).map((x: any) => x.calls)).toFixed(2)
+          : undefined,
+    },
+    {
+      id: QUERY_PERFORMANCE_REPORT_TYPES.SLOWEST_EXECUTION,
+      label: 'Slowest execution',
+      description: 'Lists queries ordered by their maximum execution time',
+      isLoading: isLoadingMMF,
+      max:
+        (slowestExecutionTime ?? []).length > 0
+          ? Math.max(...(slowestExecutionTime ?? []).map((x: any) => x.max_time)).toFixed(2)
+          : undefined,
+    },
+  ]
+
+  return (
+    <>
+      <Tabs_Shadcn_
+        defaultValue={page}
+        onValueChange={(value) => {
+          setPage(value as QUERY_PERFORMANCE_REPORT_TYPES)
+          const { sort, search, ...rest } = router.query
+          router.push({ ...router, query: { ...rest, preset: value } })
+        }}
+      >
+        <TabsList_Shadcn_ className={cn('flex gap-0 border-0 items-end')}>
```

Comment:
A better approach would be to use `NavMenu` and `NavMenuItem`. They're already used in https://supabase.com/dashboard/project/_/settings/vault/keys.

Reply:
I disagree, those components do not use tabs correctly

Reply:
I agree with the concept of wrapping them though, but the NavMenu wrap is in this case the wrong one _I think_ as the radix tabs as primitives would mean you get keyboard "left" and "right" nav

Reply:
use whatever you like for now @joshenlim @jordienr @ivasilov 
might update it later to different primitives, or offer a Tab a11y version